bin/job

```diff
@@ -31,117 +32,83 @@ main() {
   fi
 
   [[ "$SRCTYPE" = "disk" ]] && SRC="$(readlink -f $SRC)"
-  [[ "$SRCTYPE" = "libvirt" && -z "$LIBVIRTSRCURI" ]] && die "Please provide a libvirt uri"
+  [[ "$SRCTYPE" = "libvirt" && -z "$SRCURI" ]] && die "Please provide valid uri"
+  [[ "$SRCTYPE" = "libvirt" && "$SRC" =~ ^vpx && -z "$SRCPASS" ]] && die "Please provide password"
 
-  transformVM "$SRCTYPE" "$SRC" "$LIBVIRTSRCURI"
-  createEntities
+  transformVM "$SRCTYPE" "$SRC" "$SRCURI" "$SRCPASS"
 }
 
 transformVM() {
   local SRCTYPE=$1
   local SRC="$2"
-  local LIBVIRTSRCURI="$3"
+  local SRCURI="$3"
+  local SRCPASS="$4"
   local WD="$(basename $SRC).d"
 
   echo "  Converting source: $SRC"
   mkdir -p $WD
 
-  ls -shal $SRC
+  ls -shal $WD
 
   local NAME=$(basename $SRC | sed "s/[^a-zA-Z0-9-]/-/g" | tr "[:upper:]" "[:lower:]")
 
   #
   # generate disk and domxml
-  [[ "$LIBVIRTSRCURI" ]] && LIBVIRTSRCURI="-ic $LIBVIRTSRCURI"
-  virt-v2v -i "$SRCTYPE" $LIBVIRTSRCURI "$SRC" -o local -on $NAME -oa sparse -of raw -os $WD --machine-readable
-
-  local DOMXML=$(ls $WD/*.xml)
-  local VMYAML=${DOMXML%.xml}.yaml
-
-  #
-  # Convert domxml to vm.yaml
-  xsltproc $BASEDIR/data/toVMSpec.xsl $DOMXML | tee $VMYAML
-
-  if [[ $(xmllint --xpath "count(//disk[@type='file']/source)" $DOMXML) -gt 1 ]];
-  then
-    die "Only one disk per VM is supported ATM"
+  [[ "$SRCURI" ]] && SRCURI="-ic $SRCURI"
+  if [[ "SRCPASS" ]] ; then
+    echo $SRCPASS > pass
+    SRCPASS="--password-file pass"
+  fi
+  if [[ "$DEBUG" ]] ; then
+    DEBUG="-v -x"
+  fi
+  virt-v2v $DEBUG -i "$SRCTYPE" $SRCPASS $SRCURI "$SRC" -o local -on $NAME -oa sparse -of raw -os $DSTD --machine-readable
+  if [[ "SRCPASS" ]] ; then
+    rm -f pass
   fi
 
-  local DISKFILE=$(xmllint --xpath "string(//disk[@type='file']/source/@file)" $DOMXML)
-
-  #
-  # Copy to final destination
-  cp -v $DISKFILE $DSTD/disk.raw
-  cp -v $VMYAML $DSTD/vm.yaml
-
-  ls -shal $DSTD
-}
-
-createEntities() {
-  local VMYAML=$DSTD/vm.yaml
-  local DISKFILE=$DSTD/disk.raw
-
-  local PVCNAME=$(egrep "^\s+name:" <$VMYAML | tail -n1 | cut -d":" -f2 | sed "s/^\s\+//")
+  local DOMXML=$(ls $DSTD/*.xml)
+  local VMYAML=${DOMXML%.xml}.yaml
 
-  #
-  # Create the VM first, because it includes the PVC
-  #
-  kubectl create -f $VMYAML
+  # we support only one disk, it needs to be aligned with DNS-1123
+  PVCNAME=$(echo $SRC-disk-1 | sed -r 's/[_.]+/-/g')
+  VMNAME=$(echo $SRC | sed -r 's/[_.]+/-/g')
 
-  sleep "4.2s"
+  # get resources from vm xml
+  CORES=$(xmllint --xpath "/domain/vcpu/text()" $DOMXML)
+  MEMORY=$(xmllint --xpath "/domain/memory/text()" $DOMXML)
+  UNIT=$(xmllint --xpath "string(/domain/memory/@unit)" $DOMXML)
+  BUS=$(xmllint --xpath "string(/domain/devices/disk/target/@bus)" $DOMXML)
```

Comment:
Bus needs to be virtio, sata, ide, otherwise there will be an error on launch.

Reply:
OK, will fix